### PR TITLE
feat(image, layer): add bitmap mask feature

### PR DESCRIPTION
**Pull request by kisvegabor**
### Description of the feature or fix

Allow using an A8 bitmap mask for 
- the image widget
- as a style property in which case a layer will be created and that layer will be masked 

Current limitations
- In SW render transformation and mask doesn't work together
- In SW render only RGB565A8 and A8 images cannot be masked at this moment.
- The mask is always center aligned on the image. Tiling and positioning can be added later.

Test and docs is still missing. First, I would like to verify if you agree with my approach on this feature.

cc @XuNeo  

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by XuNeo** [comment]
Thank you Gabor. It looks good to me.
@bjsylvia @terry0012 @FASTSHIFT Please help to review the code.

File: docs/overview/style-props.md [hunk 1]
```diff
@@ -873,6 +873,15 @@ Set the base direction of the object. The possible values are `LV_BIDI_DIR_LTR/R
 <li style='display:inline; margin-right: 20px; margin-left: 0px'><strong>Ext. draw</strong> No</li>
 </ul>
 
+### bitmap_mask_src
+If set a layer will be created ofr the widget and the layer will be masked with this A8 bitmap mask.
```
Comment (PGNetHun) [inline docs/overview/style-props.md]: Typo: _ofr_

File: scripts/style_api_gen.py [hunk 2]
```diff
@@ -389,6 +389,9 @@
  'style_type': 'num',   'var_type': 'lv_base_dir_t', 'default':'`LV_BASE_DIR_AUTO`', 'inherited': 1, 'layout': 1, 'ext_draw': 0,
  'dsc': "Set the base direction of the object. The possible values are `LV_BIDI_DIR_LTR/RTL/AUTO`."},
 
+{'name': 'BITMAP_MASK_SRC',
+ 'style_type': 'ptr',   'var_type': 'const lv_image_dsc_t *', 'default':'`NULL`', 'inherited': 0, 'layout': 0, 'ext_draw': 0,
+ 'dsc': "If set a layer will be created ofr the widget and the layer will be masked with this A8 bitmap mask."},
```
Comment (PGNetHun) [inline scripts/style_api_gen.py]: Typo: _ofr_

File: src/draw/sw/lv_draw_sw.c [hunk 3]
```diff
@@ -264,6 +264,17 @@ static int32_t evaluate(lv_draw_unit_t * draw_unit, lv_draw_task_t * task)
...
                 }
+
+                bool transformed = draw_dsc->rotation != 0 || draw_dsc->scale_x != LV_SCALE_NONE ||
+                                   draw_dsc->scale_y != LV_SCALE_NONE ? true : false;
+
+                bool masked = draw_dsc->bitmap_mask_src != NULL;
+                if(masked && transformed)  return 0;
+
+                lv_color_format_t cf = draw_dsc->header.cf;
+                if(masked && (cf == LV_COLOR_FORMAT_A8 && cf == LV_COLOR_FORMAT_RGB565A8)) {
```
Comment (PGNetHun) [inline src/draw/sw/lv_draw_sw.c]: Typo: second `&&` operator (for color formats) should be `||`.

**Review by PGNetHun — CHANGES_REQUESTED** [review]

**Review by PGNetHun — DISMISSED** [review]

File: docs/overview/layer.rst [hunk 4]
```diff
@@ -4,6 +4,12 @@
 Layers
 ======
 
+In LVGL "layers"" can be interpreted in various ways:
```
Comment (PGNetHun) [inline docs/overview/layer.rst]: typo: double double-quotes

**Review by PGNetHun — DISMISSED** [review]

File: docs/overview/layer.rst (hunk 4 above)
Comment (kisvegabor) [reply docs/overview/layer.rst]: fixed

**Review by PGNetHun — APPROVED** [review]

**Comment by kisvegabor** [comment]
@XuNeo I merge it now, please open an issue if feel like something is missing.

**Pull request merged**